intranet/apps/eighth/views/attendance.py

```diff
@@ -678,3 +680,41 @@ def sponsor_schedule_widget_view(request):
     context.update({"eighth_sponsor": eighth_sponsor})
 
     return render(request, "eighth/sponsor_widget.html", context)
+
+
+@login_required
+@deny_restricted
+def email_students_view(request, scheduled_activity_id):
+    scheduled_activity = get_object_or_404(EighthScheduledActivity, id=scheduled_activity_id)
+
+    if not scheduled_activity.user_is_sponsor(request.user) and not request.user.is_eighth_admin:
+        raise Http404
+
+    if request.method == "POST" and request.POST.get("body"):
+        emails = [signup.user.notification_email for signup in scheduled_activity.eighthsignup_set.all()]
+
+        subject = settings.EMAIL_SUBJECT_PREFIX + "{}: Message from {}".format(scheduled_activity, request.user.full_name)
+        if request.POST.get("subject"):
+            subject += ": " + request.POST["subject"]
+
+        body_text = """{} has requested to send you the following email regarding {}:
+
+{}""".format(
+            request.user.full_name, scheduled_activity.activity, request.POST["body"]
+        )
+
+        msg = EmailMessage(
+            subject=subject, body=body_text, from_email=settings.EMAIL_FROM, to=[], reply_to=[request.user.notification_email], bcc=emails,
+        )
+
+        if settings.PRODUCTION or settings.FORCE_EMAIL_SEND:
+            msg.send()
```

Comment:
Elsewhere, we use Celery tasks to send email. I think we should do this here, especially considering some activities sponsors want to send to have large capacities.